packages/db-objects-redis/lib/objects/objectsInRedisClient.js

```diff
@@ -305,6 +305,27 @@ class ObjectsInRedisClient {
                 this.subSystem = new Redis(this.settings.connection.options);
                 this.subSystem.ioBrokerSubscriptions = {};
 
+                if (typeof this.settings.primaryHostLost === 'function') {
+                    try {
+                        // enable Expiry/Evicted events in server - same as states (could be same db)
+                        await this.client.config('set', ['notify-keyspace-events', 'Exe']);
+                    } catch (e) {
+                        this.log.warn(
+                            `${this.namespace} Unable to enable Expiry Keyspace events from Redis Server: ${e.message}`
+                        );
+                    }
+
+                    this.subSystem.on('message', (channel, message) => {
+                        if (channel === `__keyevent@${this.settings.connection.options.db}__:expired`) {
```

Comment:
if you do that, also do "evicted" :-)

Reply:
handling is 100% the same because key got deleted (just not because expiry but because of memory issues)